src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java

```diff
@@ -175,50 +175,53 @@ public ExpressionAtlasContentService(
-                            (MicroarrayExperiment) experiment, contentType) :
-                    externalResourceType.equals("ega") ?
-                    microarrayExperimentExternallyAvailableContentServiceEga.list(
-                            (MicroarrayExperiment) experiment, contentType) :
-                    microarrayExperimentExternallyAvailableContentServiceEna.list(
-                            (MicroarrayExperiment) experiment, contentType);
-            list.addAll(list2);
-            return list;
+        List arrayExpressLinks;
+        List otherExternalResourceLinks;
+        switch (experiment.getType()) {
+            case PROTEOMICS_BASELINE:
+                arrayExpressLinks = proteomicsBaselineExperimentExternallyAvailableContentService.list(
+                        (BaselineExperiment) experiment, contentType);
+                break;
+            case RNASEQ_MRNA_BASELINE:
+                arrayExpressLinks = rnaSeqBaselineExperimentExternallyAvailableContentService.list(
+                        (BaselineExperiment) experiment, contentType);
+                otherExternalResourceLinks = externalResourceType.equals("geo") ?
+                        rnaSeqBaselineExperimentExternallyAvailableContentServiceGeo.list(
+                                (BaselineExperiment) experiment, contentType) :
+                        externalResourceType.equals("ega") ?
+                                rnaSeqBaselineExperimentExternallyAvailableContentServiceEga.list(
+                                        (BaselineExperiment) experiment, contentType) :
+                                rnaSeqBaselineExperimentExternallyAvailableContentServiceEna.list(
+                                        (BaselineExperiment) experiment, contentType);
+                arrayExpressLinks.addAll(otherExternalResourceLinks);
+                break;
+            case RNASEQ_MRNA_DIFFERENTIAL:
+                arrayExpressLinks = rnaSeqDifferentialExperimentExternallyAvailableContentService.list(
+                        (DifferentialExperiment) experiment, contentType);
+                otherExternalResourceLinks = externalResourceType.equals("geo") ?
+                        rnaSeqDifferentialExperimentExternallyAvailableContentServiceGeo.list(
+                                (DifferentialExperiment) experiment, contentType) :
+                        externalResourceType.equals("ega") ?
+                                rnaSeqDifferentialExperimentExternallyAvailableContentServiceEga.list(
+                                        (DifferentialExperiment) experiment, contentType) :
+                                rnaSeqDifferentialExperimentExternallyAvailableContentServiceEna.list(
+                                        (DifferentialExperiment) experiment, contentType);
+                arrayExpressLinks.addAll(otherExternalResourceLinks);
+                break;
+            default:
```

Comment:
Usually, when we deal with switches and experiment types, we usually take this approach (remember that you can aggregate cases and this is valid syntax):
```java
case MICROARRAY_1COLOUR_MRNA_DIFFERENTIAL:
case MICROARRAY_2COLOUR_MRNA_DIFFERENTIAL:
case MICROARRAY_1COLOUR_MICRORNA_DIFFERENTIAL:
    // Do actions for microarray experiments.
    ...
    break;
default:
    throw new IllegalArgumentException(experiment.getType + ": experiment type not supported.");
    // Or maybe do nothing, and the method will return an empty list if you think that’s better.
```

See, for instance, `ExperimentDesignFileWriter` or `ExperimentSorter`.